store/src/migrations.rs

```diff
@@ -3,7 +3,46 @@ use rusqlite_migration::{Migrations, M};
 
 pub static MIGRATIONS: Lazy<Migrations> = Lazy::new(|| {
     Migrations::new(vec![M::up(
-        "CREATE TABLE nullifiers (id INTEGER PRIMARY KEY, nullifier BLOB, block_number INTEGER);",
+        "
+        CREATE TABLE
+            block_header
+        (
+            prev_hash BLOB NOT NULL,
+            block_num INTEGER NOT NULL,
+            chain_root BLOB NOT NULL,
+            account_root BLOB NOT NULL,
+            nullifier_root BLOB NOT NULL,
+            note_root BLOB NOT NULL,
+            batch_root BLOB NOT NULL,
+            proof_hash BLOB NOT NULL,
+            version INTEGER NOT NULL,
+            timestamp INTEGER NOT NULL,
+
+            PRIMARY KEY (block_num),
+            -- can not validate prev_hash value, because blocks can be pruned
+            CONSTRAINT block_header_prev_hash_valid_digest CHECK (length(prev_hash) = 32),
+            CONSTRAINT block_header_block_num_positive CHECK (block_num >= 0),
+            CONSTRAINT block_header_chain_root_valid_digest CHECK (length(chain_root) = 32),
+            CONSTRAINT block_header_account_root_valid_digest CHECK (length(account_root) = 32),
+            CONSTRAINT block_header_nullifier_root_valid_digest CHECK (length(nullifier_root) = 32),
+            CONSTRAINT block_header_note_root_valid_digest CHECK (length(note_root) = 32),
+            CONSTRAINT block_header_batch_root_valid_digest CHECK (length(batch_root) = 32),
+            CONSTRAINT block_header_proof_hash_valid_digest CHECK (length(proof_hash) = 32),
+            CONSTRAINT block_header_timestamp_positive CHECK (timestamp >= 0)
+        ) STRICT, WITHOUT ROWID;
```

Comment:
I would probably simplify this table to be just: `(block_num, header)` with `block_num` being a primary key. In the future, we could add other "meta" properties to this - to make it more like `(block_num, hash, sub_hash, header)`.

Reply:
changed